File: app/src/main/java/com/starsep/sokoban/view/SquareButton.java
package com.starsep.sokoban.view;

import android.content.Context;
import android.widget.Button;

public class SquareButton extends Button {
	public SquareButton(Context context) {
		super(context);
	}

	@Override
	protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
		int size = Math.max(widthMeasureSpec, heightMeasureSpec);
		super.onMeasure(
				MeasureSpec.makeMeasureSpec(size, MeasureSpec.EXACTLY),
				MeasureSpec.makeMeasureSpec(size, MeasureSpec.EXACTLY)
		);
	}
}

File: app/src/main/java/com/starsep/sokoban/release/view/SquareButton.kt
package com.starsep.sokoban.release.view

import android.content.Context
import com.google.android.material.button.MaterialButton
import kotlin.math.max

class SquareButton(context: Context) : MaterialButton(context) {
    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        val size = max(widthMeasureSpec, heightMeasureSpec)
        super.onMeasure(
                MeasureSpec.makeMeasureSpec(size, MeasureSpec.EXACTLY),
                MeasureSpec.makeMeasureSpec(size, MeasureSpec.EXACTLY)
        )
    }
}
